### DISCUSSION TITLE
useSWRConfig mutate function causes to request data from the api twice

### DISCUSSION THREAD

**Discussion by yyuurreecc5** [vercel/swr#2488] [Q&A]
# Bug report

## Description / Observed Behavior

When I'm calling mutate function from useSWRConfig hook the api request is called twice.

## Expected Behavior

I expect it will be called only once after mutate function is called

## Repro Steps / Code Example

[CodeSandbox](https://codesandbox.io/s/old-star-1l57c8?file=/src/App.js)

When show === true (component mounted). The api request is executed twice on one click of the revalidate button (why?). And isValidating flag changes on true only when second api request is starting. It behavior is repeated on every click.

When show === false (component unmounted). After click on the revalidate button apiRequest is executed despite the fact that there is not a single mounted component using swr on page. Is this behavior correct?

## Additional Context

SWR version: 2.1.0

**Comment by koba04** [comment]
In this case, you don't have to pass `apiRequest` to `mutate`.  `mutate(REQUEST_KEY)` works fine in your case.
https://swr.vercel.app/docs/mutation#revalidation

**Comment by yyuurreecc5** [comment]
But, then we return to the previous issue:

https://github.com/vercel/swr/issues/2416

**Reply by koba04** [reply]
That's an expected behavior.

The two requests are 
1. the `data` function in `mutate`
2. revalidation by `useSWR`

If you don't want to make revalidation by `useSWR`, you can pass `revadalite: false` as an mutate's option.

```ts
mutate(REQUEST_KEY, apiRequest, { revalidate: false });
```

**Reply by yyuurreecc5** [reply]
Yeah, it's fixing double request issue, but:
 
1) I can't use isValidating flag and as a result I can't to show in the interface that data is updating. 

2) When show === false (component unmounted). After click on the revalidate button apiRequest is executed despite the fact that there is not a single mounted component using swr on page.

[Codesandbox](https://codesandbox.io/s/elegant-river-8nxwne?file=/src/App.js)
